### DISCUSSION TITLE
EFI - Font size

### DISCUSSION THREAD

**Discussion by Tualua** [ipxe/ipxe#1015] [Q&A]
Hello!

Is there any way to increase font size? Everything is EXTREMELY small on 4K displays

**Comment by NiKiZe** [comment]
You can set different resolutions, see https://ipxe.org/cmd/console

Font support, not something that currently is user settable that I know of.

**Comment by Tualua** [comment]
Yes, I know about resolution but it doesn't help much.
Is there anything in the code that can be changed to increase font size?

**Reply by NiKiZe** [reply]
You need to actually have a font, and then care much be taken about size for each char when rendering etc.

I do think that currently fonts in firmware is used with maybe a fallback. But I didn't have time to check more than that.

But yes, I agree that a DPI aware font rendering if possible would be nice to have. (I use 4K displays as well)

**Comment by mcb30** [comment]
iPXE uses the underlying firmware's fixed-width bitmap font, which will always be eight pixels wide.

It would be possible to add support for a font scale into iPXE, to allow the font to be scaled by an integer factor (e.g. x2) in both directions.

However, since there is no support for anything more sophisticated than a linear framebuffer in iPXE, you will find that scrolling text in a 4K display resolution is very slow.  You are therefore likely to get a better experience by using the `console` command to explicitly select a lower graphics resolution.
